doc: Bluetooth: Mesh: Docs automated DFU meta gen (#14078)
nrfconnect/sdk-nrf: doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst

.. _bluetooth_mesh_dfu_eval_md:

Generating the firmware metadata
********************************

Automated metadata generation
=============================

> The most efficient way to generate the required DFU metadata is to use the DFU metadata extraction script integrated in the NCS build system. By enabling the ``CONFIG_BT_MESH_DFU_METADATA_ON_BUILD`` option in the

Reviewer: Every new sentence should be on a new line and there is no limit on the line length (ncs requirement, it is opposite in zephyr).

> application, the metadata will be automatically parsed from the .elf and .config file. The parsed data is put into a DFU zip-archive in the build folder of the application in a JSON format. Additionally, the

Reviewer: We can say the archive and file name. Other FOTA pages uses the archive name as well.
```suggestion
application, the metadata will be automatically parsed from the .elf and .config file. The parsed data is stored in `ble_mesh_metadata.json` and put into a `dfu_application.zip` archive in the build folder of the application. Additionally, the
```

Author: @Balaklaka While you are correct, I think it just becomes more confusing if we add `dfu_application.zip_ble_mesh_metadata.json`. I think it is easier if the user just have to focus on what is inside the zip.

> metadata string, required by the ``mesh models dfu metadata encode`` command, will be printed in the command line window when the application is built.

Reviewer: > required by the ``mesh models dfu metadata encode``

It is required by `dfu slot add`:
https://github.com/nrfconnect/sdk-nrf/blob/b962460658b241014e4c6274e03cb52f2b1d5ed6/doc/nrf/protocols/bt/bt_mesh/dfu_over_bt_mesh.rst?plain=1#L52-L58

Author: Right... That is what I meant to write 👍

> The most efficient way to generate the required DFU metadata is to use the DFU metadata extraction script integrated in the NCS build system. By enabling the ``CONFIG_BT_MESH_DFU_METADATA_ON_BUILD`` option in the
> application, the metadata will be automatically parsed from the .elf and .config file. The parsed data is put into a DFU zip-archive in the build folder of the application in a JSON format. Additionally, the
> metadata string, required by the ``mesh models dfu metadata encode`` command, will be printed in the command line window when the application is built.

Reviewer 2: ```suggestion
By enabling the :kconfig:option:`CONFIG_BT_MESH_DFU_METADATA_ON_BUILD` option in the application, the metadata will be automatically parsed from the ``.elf`` and ``.config`` files.
The parsed data is stored in the :file:`ble_mesh_metadata.json` file.
The file is placed in the :file:`dfu_application.zip` archive in the build folder of the application.
Additionally, the metadata string required by the ``mesh models dfu slot add`` command will be printed in the command line window when the application is built.
```

> .. note::
>    Currently this script only supports applications that contain a single composition data structure, meaning that applications containing multiple composition data structures must rely on manual generation of
>    the metadata string(s). Additionally, the script is hardcoded to produce a metadata string where the firmware is targeted for the application core.

Reviewer: ```suggestion
.. note::
   Currently this script only supports applications that contain a single composition data structure, meaning that applications containing multiple composition data structures must rely on manual generation of the metadata string(s).
   It is also required that the composition data is declared with const-qualifier.
   Additionally, the script is hardcoded to produce a metadata string where the firmware is targeted for the application core.
```

Reviewer: This part is missing:
> It is also required that the composition data is declared with const-qualifier.

> 
> In addition to printing the metadata string on build, a separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system. This provides the user
> easy access to this information without having to enter the .json file in the build folder or to rebuild the application::

Reviewer 2: ```suggestion

In addition to printing the metadata string when the application is built, a separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system.
This gives the user easy access to this information, without having to enter the .json file in the build folder or to rebuild the application::
```

  west ble-mesh-dfu-metadata

> For this particular example, the output generated by either the build system or the west ble-mesh-dfu-metadata command would be::

Reviewer 2: ```suggestion
For this particular example, the output generated by either the build system or the west ``ble-mesh-dfu-metadata`` command would be::
```

>   Encoded metadata: 020000000100000094cf24017c26f3710100

Reviewer: The complete script output should be added here:
https://github.com/nrfconnect/sdk-nrf/blob/7d209a57228062b2453d8ba25b172e8d09a26d49/scripts/bluetooth/mesh/mesh_dfu_metadata.py#L408-L410

Automated metadata generation
=============================

By enabling the :kconfig:option:`CONFIG_BT_MESH_DFU_METADATA_ON_BUILD` option in the application, the metadata will be automatically parsed from the ``.elf`` and ``.config`` files.
The parsed data is stored in the :file:`ble_mesh_metadata.json` file.
The file is placed in the :file:`dfu_application.zip` archive in the build folder of the application.
Additionally, the metadata string required by the ``mesh models dfu slot add`` command will be printed in the command line window when the application is built::

  Bluetooth Mesh Composition metadata generated:
    Encoded metadata: 020000000100000094cf24017c26f3710100
    Full metadata written to: APPLICATION_FOLDER\build\zephyr\dfu_application.zip

.. note::
   Currently this script only supports applications that contain a single composition data structure, meaning that applications containing multiple composition data structures must rely on manual generation of the metadata string(s).
   Additionally, the script is hardcoded to produce a metadata string where the firmware is targeted for the application core.

> In addition to printing the metadata string when the application is built, a separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system.

Reviewer: The last part of the sentence tells the same as the first one. I'd suggest to remove the first part of the sentence:
```suggestion
A separate west command can be utilized to print the metadata to the console, given that it is already generated by the build system.
```
